src/Bicep.Core/Parser/Parser.cs

```diff
@@ -240,9 +240,15 @@ private SyntaxBase MemberExpression()
                 {
                     // dot operator
                     Token dot = this.reader.Read();
-                    IdentifierSyntax propertyName = this.Identifier(b => b.ExpectedPropertyIdentifier());
```

Comment:
Can we just consume the identifier here and pass it as an argument to `InstanceFunctionCall` or `PropertyAccess`? We should then be able to replace `IsFunctionCall()` with just a `reader.Check(...)`